enhancements/ingress/lb-allowed-source-ranges.md

```diff
@@ -0,0 +1,227 @@
+based on `AllowedSourceRanges`.
+
+#### Variation [optional]
+
+N/A
+
+### API Extensions
+
+This proposal will modify the `IngressController` API by adding a new 
+variable called `AllowedSourceRanges` to the `LoadBalancerStrategy` 
+struct type and a new type `CIDR` that is a string in CIDR notation.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The value set for `AllowedSourceRanges` will be used to set `.Spec.LoadBalancerSourceRanges` 
+of the LoadBalancer Service in front of the router in cloud environments.
+
+If the customer directly sets `.Spec.LoadBalancerSourceRanges` of the LoadBalancer Service,
+its value will be used to set `AllowedSourceRanges` if and only if `AllowedSourceRanges`
+was not initially set. Otherwise, `.Spec.LoadBalancerSourceRanges` is overwritten by `AllowedSourceRanges`.
+
+### Risks and Mitigations
+
+One risk of this proposal is that customers may have already set the
+`.Spec.LoadBalancerSourceRanges` field of the LoadBalancer Service. 
+This risk is mitigated by using this value to set `AllowedSourceRanges` 
+if and only if `AllowedSourceRanges` was not initially set.
+
+### Drawbacks
+
+This enhancement may bring additional complexity for the customers who are 
+used to modify directly the `Spec.LoadBalancerSourceRanges` of the LoadBalancer 
+Service for the same purposes. 
+
+## Design Details
+
+### Open Questions [optional]
+
+1. On Kubernetes API reference, the field 
+[`loadBalancerSourceRanges`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#service-v1-core) 
+has the warning "This field will be ignored if the cloud-provider does not support the feature.".
+However, there is no information about which platforms support this. Is there any platform
+OpenShift runs on which does not support this feature?
+
+### Test Plan
+
+The operator that manages the IngressController Deployment and related
+resources has unit test coverage; for this enhancement, the unit tests are
+expanded to cover the additional functionality.  
+
+The operator has end-to-end tests; for this enhancement, a test is added that verifies
+the following scenarios:
+
+1. Adding/deleting/changing `AllowedSourceRanges` on the IngressController and observing `.Spec.LoadBalancerSourceRanges` on the LoadBalancer Service.
+
+2. Adding `.Spec.LoadBalancerSourceRanges` to a LoadBalancer Service and observing `AllowedSourceRanges` on the IngressController in which it was unset previously.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A. This feature will go directly to GA.
+
+#### Tech Preview -> GA
+
+N/A. This feature will go directly to GA.
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+If the customer is directly setting `.Spec.LoadBalancerSourceRanges` of
+the Load Balancer Service, this value will be used to set `AllowedSourceRanges`
+during the upgrade.
```

Comment:
Are you proposing that the operator would set `spec.allowedSourceRanges`?  The operator shouldn't be updating `spec`; the user owns `spec`, and the operator owns `status`.  Anyway, how would the operator distinguish between (a) the user upgraded and didn't set `spec.allowedSourceRanges` and (b) the user upgraded and later set `spec.allowedSourceRanges` to empty?

Reply:
This idea is dropped as we have another migration workflow now.